src/zenml/entrypoints/base_entrypoint_configuration.py

```diff
@@ -214,12 +214,14 @@ def download_code_if_necessary(
         if not should_download_code:
             return
 
-        if code_reference := deployment.code_reference:
+        if code_path := deployment.code_path:
+            code_utils.download_code_from_artifact_store(code_path=code_path)
+        elif code_reference := deployment.code_reference:
+            # TODO: This might fail if the code repository had unpushed changes
```

Comment:
Reminder for the TODO.

Reply:
I would not tackle that as part of this PR. I think it's very unlikely for people to actually run into this (I haven't heard of it ever), and we're planning a bigger rework of code repositories anyway when this might or might not be relevant anymore. WDYT?

Reply:
Makes sense. Just wanted to leave the comment as a reminder in case it slipped through the cracks. Feel free to resolve it.